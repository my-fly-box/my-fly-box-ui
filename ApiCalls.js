export const fetchFlies = async () => {
  try {
    const response = await fetch(
      "https://my-fly-box-api.herokuapp.com/api/v1/flies"
    );
    const flyData = await response.json();
    return flyData;
  } catch (error) {
    console.error(error);
  }
};

export const addFlyToAPI = async (flyData) => {
  try {
    const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/flies', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
};

export const addUpdatedFly = async (flyData) => {
  try {
    const response = await fetch(`https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyData.id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData.attributes),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
}
export const deleteFly = async (flyId) => {
    try {
      const response = await fetch(
        `https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyId}`,
        {
          method: "DELETE",
          headers: {
            "Content-Type": "application/json;charset=utf-8",
          },
        }
      );
      const deletionResponse = await response.json();
      console.log(deletionResponse);
    } catch (error) {
      console.error(error);
    }
  };

  export const addFishToAPI = async (fishData) => {
    try {
      const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/catches', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json;charset=utf-8'
        },
        body: JSON.stringify(fishData),
      });
      const fishDataResponse = await response.json();
      return fishDataResponse;
    } catch (error) {
      console.error(error);
    }
  };
