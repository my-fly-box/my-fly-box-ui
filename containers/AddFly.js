import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import FlyForm from '../components/FlyForm'

export default function AddFly() {
    return (
        <View style={styles.container}>
            <View>
                <FlyForm />
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 45,
    },
})