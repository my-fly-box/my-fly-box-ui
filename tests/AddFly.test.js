import React from 'react';
import 'react-native'
import AddFly from '../containers/AddFly'
import { Provider } from 'react-redux'
import rootReducer from '../reducers/index'
import { createStore } from 'redux'
import renderer from 'react-test-renderer'
import { fetchFlies } from '../ApiCalls'

jest.mock('../ApiCalls')


describe('AddFly', () => {
  let mockFlyData
  beforeEach(() => {
    mockFlyData = [
      {
        "id": "18",
        "type": "fly",
        "attributes": {
            "id": 18,
            "name": "Copper John",
            "size": 16,
            "color": "Copper",
            "category": "Nymph",
            "amount": 3
        }
    }
    ]
  })

  it('renders the AddFly', async () => {
    const testStore = createStore(rootReducer)
    const testWrapper = <Provider store={testStore}><AddFly /></Provider>
    fetchFlies.mockResolvedValueOnce(mockFlyData)
    const snap = renderer.create(
      testWrapper
    ).toJSON();
    expect(snap).toMatchSnapshot();
  })
}) 