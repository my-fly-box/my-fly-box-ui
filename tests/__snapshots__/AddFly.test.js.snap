// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFly renders the AddFly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f8f8ff",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": 45,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#0b7d83",
        "flex": 0.2,
        "fontFamily": "Helvetica Neue",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Please enter the fly's information:
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": 45,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Fly Name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 18,
          "height": "25%",
          "marginBottom": "3%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Fly Color"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 18,
          "height": "25%",
          "marginBottom": "3%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Fly Size"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 18,
          "height": "25%",
          "marginBottom": "3%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter Fly Category"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 18,
          "height": "25%",
          "marginBottom": "3%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter Fly Amount"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "color": "#212326",
          "elevation": 3,
          "fontSize": 18,
          "height": "25%",
          "marginBottom": "3%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "textAlign": "center",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f7841f",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica-Bold",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f7841f",
          "borderColor": "#212326",
          "borderRadius": 5,
          "borderWidth": 0.2,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica-Bold",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
